### DISCUSSION TITLE
A simplified scriptlet to remove conditional `cap_html_filtering` elements

### DISCUSSION THREAD

**Discussion by K-mikaZ** [uBlockOrigin/uBlock-issues#3313] [Ideas]
I think some filters could be simplified to a simple function call in a scriptlet.

use:
```
{{domain}}##+js(remove-html-element, {{element}})
```
which would simplify the function:
```
!#if cap_html_filtering
{{domain}}##^{{element}}
!#else
{{domain}}#?#{{element}}:remove()
!#endif
```

example for:
```
numerama.com##+js(remove-html-element, #premium-subscribe-modal)
```

instead of:
```
!#if cap_html_filtering
numerama.com##^#premium-subscribe-modal
!#else
numerama.com#?##premium-subscribe-modal:remove()
!#endif
```

And maybe consider specific filters like:
```
numerama.com##^.is-distinct-mite:has(button[data-dialog-controls*="premium-subscribe"])
```

**Comment by Yuki2718** [comment]
Only complicates, HTML filtering doesn't always work.

**Reply by krystian3w** [reply]
I see opened only https://github.com/uBlockOrigin/uBlock-issues/issues/3222

**Reply by garry-ut99** [reply]
> Yuki2718 : Only complicates, HTML filtering doesn't always work.

From your statement it seems that you have misunderstood OP's idea, as OP's scriptlet didn't meant to "drop using :remove() and to use only html filtering", his idea was to combine the both solutions into a single scriptlet, and let the scriptlet to detect and decide which solution to apply to an element.

But like gorhill already mentioned, `html filtering` and `js injecting` are different things: html filtering can't be injected by a +js scriptlet, as html filtering is done on a higher level: before any scriptlets are injected.

> krystian3w :  I see opened only https://github.com/uBlockOrigin/uBlock-issues/issues/3222

**1)** There is also https://github.com/uBlockOrigin/uBlock-issues/issues/1507. **2)** From your statement it seems that you have misunderstood Yuki's comment or have forgotten about the other html filtering limitations, because what Yuki meant, was most likely that:
- Chromium-based browsers don't support HTML filtering
- HTML filtering support ~99,5% encodings, which means it occassionally might not work
(one site per several hundred, or if we assume ~99,9% then 1 per 1000)

But like gorhill already mentioned, any discussion about injecting html filtering via scriptelsts makes no sense, because it's not possible.

**Reply by krystian3w** [reply]
Then he would completely skip the if else endif construction.

*The problem with Firefox mobile should be checked once a year to see if it is up to date and has not been there for "10 years".*

**Reply by krystian3w** [reply]
Also I do not see report that pop-up and button to EasyList or Dandelion repo.

**Comment by gorhill** [comment]
HTML filtering is meant to remove DOM elements in the response body, before the browser parses the response. Cosmetic filtering is no different than scriptlet injection in that it works on the live DOM. These two filtering approaches do not do the same thing, they can't be conflated.

**Comment by krystian3w** [comment]
> ```adb
> numerama.com##^.is-distinct-mite:has(button[data-dialog-controls*="premium-subscribe"])
> ```

Smell self-promo. I suggest ask EasyList Annoyances to consider address for more than 1 human in custom filters (https://github.com/easylist/easylist/issues).
If the application is rejected, you have to check other Annoyances lists, such as: Dandelion Annoyances https://filterlists.com/lists/dandelion-sprouts-annoyances-list or if you browse this website on multiple devices, it's your own GitHub Gist (https://gist.github.com/ - ok for start as private file) or GitHub repository (better to share with big range, allow other people with write access (e.g. from another continent to increase responsible repairs time when you goes to job or sleep) and can be used GitHub Actions).

*If you see bug with use CSS4 `:has` selector in HTML Filtering then should be reported as issue (to assess whether it is possible to fix or enforce the use of producedural syntax maybe by `:upward`).* Poor filter should be updated to avoid glitches in Firefox 121-127: https://github.com/uBlockOrigin/uAssets/discussions/23949 (or try reporting it to the browser developers with a decent description).
